tfe/resource_tfe_policy_set.go

```diff
@@ -22,6 +23,17 @@ func resourceTFEPolicySet() *schema.Resource {
 			"name": {
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errs []error) {
+					value := v.(string)
+					matched, err := regexp.Match(`^[A-Za-z\d-_]+$`, []byte(value))
```

Comment:
As described here https://www.terraform.io/docs/cloud/api/policy-sets.html 
![Screenshot_2020-05-23_10-48-47](https://user-images.githubusercontent.com/777823/82726260-0befde00-9ce3-11ea-9f1b-bca2b9beca0d.png)

Reply:
Thanks for submitting this PR. Having reviewed the contents I was wondering if any consideration was given to using one of the validators that are available in the [validation package](https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/validation#pkg-index)? 

 I think the use of the [StringMatch](https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/validation#StringMatch) validator will simplify things a great deal. As an example:

```
"name": {
	Type:         schema.TypeString,
	Required:     true,
	ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[\w\.\-]+$`), "The name of the policy set. Can only include letters, numbers, -, and _."),
},
```

Reply:
Yup, makes sense.  I'll adjust.  Thanks

Reply:
Updated the validation function to the native validation call and updated the test to match the new error message

Reply:
Tests still show passing

```
$ TESTARGS="-run TestAccTFEPolicySet_invalidname" make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEPolicySet_invalidname -timeout 15m
?   	github.com/terraform-providers/terraform-provider-tfe	[no test files]
=== RUN   TestAccTFEPolicySet_invalidname
--- PASS: TestAccTFEPolicySet_invalidname (0.01s)
PASS
ok  	github.com/terraform-providers/terraform-provider-tfe/tfe	(cached)
?   	github.com/terraform-providers/terraform-provider-tfe/version	[no test files]
```